src/libraries/Microsoft.PowerFx.Interpreter/RecalcEngine.cs

```diff
@@ -215,40 +216,10 @@ internal void DefineType(string script, ParserOptions parserOptions)
             foreach (var defType in definedTypes)
             {
                 var name = defType.Ident.Name.Value;
-                var res = DTypeFromTexlNode(defType.Type.TypeRoot) ?? throw new Exception("Failed defining type");
-                _definedTypeSymbolTable.RegisterType(name, new KnownRecordType(res));
-            }
-        }
-
-        internal DType DTypeFromTexlNode(TexlNode node)
-        {
-            if (node is RecordNode recordNode)
-            {
-                return DTypeFromRecordNode(recordNode);
-            }
-            else if (node is FirstNameNode nameNode)
-            {
-                return GetTypeFromName(nameNode.Ident.Name.Value);
-            }
+                var res = defType.Type.TypeRoot.Accept(new DTypeVisitor(), _definedTypeSymbolTable) ?? throw new Exception("Failed defining type");

```

Comment:
Should this throw, or should it add an error?

Reply:
Also, probably DefineType should return a *Result type instead of void, so that we can get errors out ...

Reply:
Again, not for this PR, future notes